File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.py
import sys
input = sys.stdin.readline

n = int(input())
a=[]
for i in range(n):
    x,y = map(int,input().split())
    a.append([x,y])

a.sort(key=lambda y: y[1])
a.sort(key=lambda x: x[0])


for x in a:
    print(*x)

File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
});

rl.on("close", function () {
  const n = Number(input[0]);
  let arr = [];
  for (let i = 0; i < n; i++) {
    const [x, y] = input[i + 1].split(" ").map(Number);
    arr.push([x, y]);
  }

  arr.sort((a, b) => {
    if (a[0] === b[0]) {
      return a[1] - b[1];
    } else {
      return a[0] - b[0];
    }
  });
  const output = arr.map((i) => i.join(" ")).join("\n");
  console.log(output);
});
